File: 1507.reformat-date.cpp
/*
 * @lc app=leetcode id=1507 lang=cpp
 *
 * [1507] Reformat Date
 */
#include <string>
#include <unordered_map>

using namespace std;

// @lc code=start
class Solution {
public:
    string reformatDate(string date) {
        unordered_map<string, string> months = {
            {"Jan", "01"},
            {"Feb", "02"},
            {"Mar", "03"},
            {"Apr", "04"},
            {"May", "05"},
            {"Jun", "06"},
            {"Jul", "07"},
            {"Aug", "08"},
            {"Sep", "09"},
            {"Oct", "10"},
            {"Nov", "11"},
            {"Dec", "12"}
        };

        return date.length() == 13 ? 
            date.substr(9,4) + '-' + months[date.substr(5,3)] + '-' + date.substr(0,2) : 
            date.substr(8,4) + '-' + months[date.substr(4,3)] + '-' + '0' + date.substr(0,1);
    }
};
// @lc code=end

File: 1507.reformat-date.java
/*
 * @lc app=leetcode id=1507 lang=java
 *
 * [1507] Reformat Date
 */
import java.util.*;

// @lc code=start
class Solution {
    private static final Map<String, String> MONTH = new HashMap<String, String>() {{
        put("Jan", "01");
        put("Feb", "02");
        put("Mar", "03");
        put("Apr", "04");
        put("May", "05");
        put("Jun", "06");
        put("Jul", "07");
        put("Aug", "08");
        put("Sep", "09");
        put("Oct", "10");
        put("Nov", "11");
        put("Dec", "12");
    }};;

    public String reformatDate(String date) {
        String[] data = date.split("\\s+");
        StringBuilder sb = new StringBuilder();
        sb.append(data[2]).append('-');     // year
        sb.append(MONTH.get(data[1])).append('-');  // month
        // Add a leading zero if day is less than 10 
        sb.append(data[0].length() == 3 ? ('0' + data[0].substring(0, 1)) : data[0].substring(0, 2));
        return sb.toString();
    }
}
// @lc code=end

